### PR TITLE
Fix issue

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

<<if there's an issue>>Fixes #issueNumber

*High-level description of what your changes are accomplishing*

*Add screenshots if applicable. Images are nice :)*

  # To Test

1. <<Step one>> Open the page `about`
- [ ] <<What to expect?>> Verify it contains about information...
- [ ] Checkbox Style list of things a QA person could verify, i.e.
- [ ] Should display Text Input our storybook
- [ ] Input should not accept Numbers
2. <<Step two>> ...

  # Background

*Optional: Give background information for changes you've made, that might be difficult to explain via comments*

### CODE REVIEW

**Comment by alfetopito** [comment]
I'll document the explanation of what's happening in this issue.

Without code changes just adding a few debugging entries, we get to this:

Screen Shot 2022-02-25 at 14 52 41


The first part is logging stuff from https://github.com/gnosis/cowswap/blob/d42aabe6f0/src/custom/state/claim/hooks/index.ts#L970-L975

This is where the total max investment amount in investment currency is calculated.
This is where the weird amount is coming from

The second part is logging stuff from
https://github.com/gnosis/cowswap/blob/d42aabe6f087ac5f304edbd8b535ad1e145a94a9/src/custom/state/claim/hooks/utils.ts#L181

This is where the source for the missing atom is coming, causing amounts to be 99.99% claimed, but not quite 100%

**Comment by alfetopito** [comment]
Here's regarding the last missing amount on the way back.

Screen Shot 2022-02-25 at 18 10 49

As can be seen for the second claim, the original claim amount is `166666666666667000000000`, while the amount we get by applying the price (0.15 USDC per vCOW) we get `166666666666666666666666`

File: src/custom/state/claim/hooks/utils.ts [hunk 1]
```diff
@@ -165,20 +165,20 @@ export function calculatePercentage<C1 extends Currency, C2 extends Currency>(
...
-    return { vCowAmount: claimAmount, investmentCost: currencyAmount }
-  } else if (!currencyAmount || !price) {
+    return { vCowAmount: claimAmount, investmentCost: claimAmount }
+  } else if (!investCurrency || !price) {
     return {}
   }
 
-  const amount = CurrencyAmount.fromRawAmount(currencyAmount.currency, investedAmount)
   return { vCowAmount: price.quote(amount), investmentCost: amount }
+  const amount = CurrencyAmount.fromRawAmount(investCurrency, investedAmount)
```
Comment (W3stside) [inline src/custom/state/claim/hooks/utils.ts:180]: ? just noting this. i am looking by commit but this shouldn't make sense with the return above

Comment (W3stside) [reply src/custom/state/claim/hooks/utils.ts:180]: nvm, now i saw it was removed 👍

File: src/custom/state/claim/hooks/utils.ts [hunk 2]
```diff
@@ -177,8 +177,8 @@ export function calculateInvestmentAmounts(
     return {}
   }
 
-  return { vCowAmount: price.quote(amount), investmentCost: amount }
   const amount = CurrencyAmount.fromRawAmount(investCurrency, investedAmount)
+  return { vCowAmount: price.invert().quote(amount), investmentCost: amount }
```
Comment (W3stside) [inline src/custom/state/claim/hooks/utils.ts:181]: but what is the difference between inverting above and now here?

Comment (W3stside) [reply src/custom/state/claim/hooks/utils.ts:181]: is the math found in the method used on line`975` fixing this? if so, damn, that's sketchy af for us to be using this math lib...

File: src/custom/state/claim/hooks/index.ts [hunk 3]
```diff
@@ -62,7 +62,7 @@ import useIsMounted from 'hooks/useIsMounted'
 import { ChainId } from '@uniswap/sdk'
 import { ClaimInfo } from 'state/claim/reducer'
 
-const CLAIMS_REPO_BRANCH = 'gip-13'
```
Comment (W3stside) [inline src/custom/state/claim/hooks/index.ts:65]: careful here about the package (just note)

**Comment by anxolin** [comment]
@alfetopito let me know if we close this one or u want it open

**Comment by alfetopito** [comment]
I was supposed to do a post-morten after finishing with the uni-merge.
Keeping it open will be better to remind something must still be done :)

**Pull request closed without merging**